File: backend/src/main/java/com/trio/backend/service/AnalyticsAIService.java
package com.trio.backend.service;

import com.trio.backend.dto.ai.AnalyticsAIEditRequest;
import com.trio.backend.dto.ai.AnalyticsAIResponse;
import com.trio.backend.enums.AIScopeType;

import java.time.LocalDate;
import java.util.UUID;

public interface AnalyticsAIService {

    AnalyticsAIResponse generate(UUID workspaceId, UUID departmentId, UUID projectId,
                                  LocalDate startDate, LocalDate endDate);

    AnalyticsAIResponse generate(UUID workspaceId, UUID departmentId, UUID projectId, UUID teamId,
                                  AIScopeType scope, LocalDate startDate, LocalDate endDate);

    AnalyticsAIResponse regenerate(UUID workspaceId, UUID departmentId, UUID projectId, UUID reportId);

    AnalyticsAIResponse edit(UUID workspaceId, UUID departmentId, UUID projectId, UUID reportId,
                              AnalyticsAIEditRequest request);

    AnalyticsAIResponse approve(UUID workspaceId, UUID departmentId, UUID projectId, UUID reportId);

    AnalyticsAIResponse reject(UUID workspaceId, UUID departmentId, UUID projectId, UUID reportId);
}

File: frontend/project/src/services/analytics-ai-service.ts
import { apiClient } from '../lib/api';

export interface AnalyticsAIGenerateRequest {
  workspaceId: string;
  departmentId?: string;
  projectId?: string;
  teamId?: string;
  scope?: 'WORKSPACE' | 'DEPARTMENT' | 'PROJECT' | 'TEAM';
  startDate?: string;
  endDate?: string;
}

export interface AnalyticsAIEditRequest {
  workspaceId: string;
  departmentId: string;
  projectId?: string;
  executiveSummary?: string;
  kpiHighlights?: string;
  trendsSummary?: string;
  riskAssessment?: string;
  recommendations?: string;
  detailedReport?: string;
}

export interface AnalyticsAIResponse {
  reportId: string;
  workspaceId: string;
  departmentId: string;
  projectId?: string;
  reportDate: string;
  timeRangeStart?: string;
  timeRangeEnd?: string;
  executiveSummary: string;
  kpiHighlights: string;
  trendsSummary: string;
  riskAssessment: string;
  recommendations: string;
  detailedReport: string;
  generationStatus: 'PENDING' | 'GENERATED' | 'FAILED';
  generationDate: string;
  generatedBy: string;
  executionTime: number;
  createdAt: string;
  updatedAt: string;
}

export function analyticsAIService() {
  const base = '/analytics/ai';
  return {
    generate: (data: AnalyticsAIGenerateRequest) =>
      apiClient.post<AnalyticsAIResponse>(`${base}/generate`, data),

    regenerate: (reportId: string, data: AnalyticsAIGenerateRequest) =>
      apiClient.post<AnalyticsAIResponse>(`${base}/regenerate/${reportId}`, data),

    edit: (reportId: string, data: AnalyticsAIEditRequest) =>
      apiClient.put<AnalyticsAIResponse>(`${base}/${reportId}`, data),

    approve: (reportId: string) =>
      apiClient.post<AnalyticsAIResponse>(`${base}/${reportId}/approve`),

    reject: (reportId: string) =>
      apiClient.post<AnalyticsAIResponse>(`${base}/${reportId}/reject`),
  };
}
